submissions/AssmaaM/Solution.py

```diff
@@ -0,0 +1,35 @@
+from collections import defaultdict
+
+def process_input(input_file):
+    city_prices = defaultdict(float)
+    city_products = defaultdict(list)
+
+    with open(input_file, 'r') as file:
```

Comment:
Will this load the whole file in memory? This might trigger an OOM

Reply:
The code reads the input file line by line in the `process_input` function, ensuring that the entire file is not loaded into memory at once